src/query/service/src/interpreters/interpreter_insert_v2.rs

```diff
@@ -101,6 +117,177 @@ impl InsertInterpreterV2 {
+                    let compression = StageFileCompression::from_str(v)?;
+                    stage.file_format_options.compression = compression;
+                }
+                "row_tag" => stage.file_format_options.row_tag = v.clone(),
+                "quote" => stage.file_format_options.quote = v.clone(),
+
+                // copy options
+                "on_error" => {
+                    let on_error = OnErrorMode::from_str(v)?;
+                    stage.copy_options.on_error = on_error;
+                }
+                "size_limit" => {
+                    let size_limit = usize::from_str(v)?;
+                    stage.copy_options.size_limit = size_limit;
+                }
+                "split_size" => {
+                    let split_size = usize::from_str(v)?;
+                    stage.copy_options.split_size = split_size;
+                }
+                "purge" => {
+                    let purge = bool::from_str(v).map_err(|_| {
+                        ErrorCode::StrParseError(format!("Cannot parse purge: {} as bool", v))
+                    })?;
+                    stage.copy_options.purge = purge;
+                }
+                "single" => {
+                    let single = bool::from_str(v).map_err(|_| {
+                        ErrorCode::StrParseError(format!("Cannot parse single: {} as bool", v))
+                    })?;
+                    stage.copy_options.single = single;
+                }
+                "max_file_size" => {
+                    let max_file_size = usize::from_str(v)?;
+                    stage.copy_options.max_file_size = max_file_size;
+                }
+
+                _ => {}
+            }
+        }
+
+        Ok(())
+    }
+
+    async fn build_insert_from_stage_pipeline(
+        &self,
+        table: Arc<dyn Table>,
+        attachment: Arc<StageAttachment>,
+        pipeline: &mut Pipeline,
+    ) -> Result<()> {
+        let start = Instant::now();
+        let ctx = self.ctx.clone();
+        let table_ctx: Arc<dyn TableContext> = ctx.clone();
+        let source_schema = self.plan.schema();
+        let target_schema = table.schema();
+        let catalog_name = self.plan.catalog.clone();
+        let overwrite = self.plan.overwrite;
+
+        let (mut stage_info, path) = parse_stage_location(&self.ctx, &attachment.location).await?;
+        self.apply_stage_options(&mut stage_info, &attachment.params)?;
+
+        let mut stage_table_info = StageTableInfo {
+            schema: source_schema.clone(),
+            user_stage_info: stage_info,
+            path: path.to_string(),
+            files: vec![],
+            pattern: "".to_string(),
+            files_to_copy: None,
+        };
+
+        let all_source_file_infos = StageTable::list_files(&table_ctx, &stage_table_info).await?;
+
+        // TODO:(everpcpc) color_copied_files
+
+        let mut need_copied_file_infos = vec![];
+        for file in &all_source_file_infos {
```

Comment:
I think this is used only for copy, which commit the paths to meta  with `set_on_finished`.
for `insert with stage_attachment.`, I think we do not need it.
the path maybe a tmp uuid.  if upload ok and insert fail, we may retry this insert (dep on the clear error code)
finally the tmp file is cleaned